peps/pep-0728.rst

```diff
@@ -0,0 +1,647 @@
+PEP: 728
+Title: TypedDict with Typed Extra Items
+Author: Zixuan James Li <p359101898@gmail.com>
+Sponsor: Jelle Zijlstra <jelle.zijlstra@gmail.com>
+Discussions-To:
+Status: Draft
+Type: Standards Track
+Topic: Typing
+Content-Type: text/x-rst
+Created: 12-Sep-2023
+Python-Version: 3.13
+Post-History:
+
+
+.. highlight:: rst
+
+Abstract
+========
+
+This PEP proposes a way to type extra items for :class:`~typing.TypedDict` using
+a reserved ``__extra__`` key. This addresses the need to define a subset of
```

Comment:
Noting for the record that I'm not completely happy with this syntax, because it feels wrong to pick one specific key to special-case. However, other possible solutions also have problems, and I'm OK with merging the PEP as is; the PEP discussion can cover possible alternatives.

If we do stick with this version, I'd prefer the shorter name `__extra__` over `__extra_keys__`. The longer name doesn't really make a collision much less likely.